<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from arm.xacro                      | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="arm">
  <material name="silver_default">
    <color rgba="0.700 0.700 0.700 1.000"/>
  </material>
  <material name="default">
    <color rgba="0.3137254901960784 0.3137254901960784 0.3137254901960784 1.0"/>
  </material>
  <material name="steel_satin">
    <color rgba="0.6274509803921569 0.6274509803921569 0.6274509803921569 1.0"/>
  </material>
  <link name="base_link">
    <!-- <inertial>
    <origin xyz="-0.002716977299773592 1.9808719046125134e-13 -0.002500000000000375" rpy="0 0 0"/>
    <mass value="0.17190979108994076"/>
    <inertia ixx="7.2e-05" iyy="7.6e-05" izz="0.000147" ixy="-0.0" iyz="0.0" ixz="-0.0"/>
  </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="steel_satin"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.1170701569437167 0.05258495071306836 0.24677061589144064"/>
      <mass value="0.6560949347888712"/>
      <inertia ixx="0.020432" ixy="0.000756" ixz="0.009271" iyy="0.024363" iyz="-0.001594" izz="0.005288"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0375 -0.034 -0.0425"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/link1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="default"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0375 -0.034 -0.0425"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/link1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link2_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0912652794492105 -0.04755997480137449 0.15715086928559246"/>
      <mass value="0.4295491979731537"/>
      <inertia ixx="0.0092" ixy="0.000233" ixz="-0.005183" iyy="0.01219" iyz="0.000401" izz="0.003285"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.235094 -0.046 -0.459007"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/link2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="default"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.235094 -0.046 -0.459007"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/link2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="tip_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.012554620120246197 2.599084142351771e-13 0.021619020423368274"/>
      <mass value="0.03924999999999999"/>
      <inertia ixx="7e-06" ixy="0.0" ixz="-3e-06" iyy="9e-06" iyz="0.0" izz="3e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.016007 0.0075 -0.891382"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/tip_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="steel_satin"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.016007 0.0075 -0.891382"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/tip_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="shoulder_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00379841512801708 -4.326319553927361e-16 0.05885490789415437"/>
      <mass value="0.2699874429831712"/>
      <inertia ixx="0.000593" ixy="0.0" ixz="-1.7e-05" iyy="0.000639" iyz="-0.0" izz="0.0003"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/shoulder_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="default"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 -0.0 0.0"/>
      <geometry>
        <mesh filename="package://arm_description/meshes/shoulder_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="Rev1" type="continuous">
    <origin rpy="0 0 0" xyz="-0.0375 0.034 0.0425"/>
    <parent link="shoulder_1"/>
    <child link="link1_1"/>
    <axis xyz="-0.0 1.0 -0.0"/>
  </joint>
  <joint name="Rev2" type="continuous">
    <origin rpy="0 0 0" xyz="-0.197594 0.012 0.416507"/>
    <parent link="link1_1"/>
    <child link="link2_1"/>
    <axis xyz="0.0 -1.0 0.0"/>
  </joint>
  <joint name="Rev3" type="revolute">
    <origin rpy="0 0 0" xyz="0.251101 -0.0535 0.432375"/>
    <parent link="link2_1"/>
    <child link="tip_1"/>
    <axis xyz="-0.0 1.0 0.0"/>
    <limit effort="100" lower="-1.22173" upper="1.22173" velocity="100"/>
  </joint>
  <joint name="Rev4" type="continuous">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="shoulder_1"/>
    <axis xyz="-0.0 0.0 1.0"/>
  </joint>
</robot>

